zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FlowNodeInstanceFromIncidentHandler.java

```diff
@@ -59,29 +67,35 @@ public FlowNodeInstanceEntity createNewEntity(final String id) {
   public void updateEntity(
       final Record<IncidentRecordValue> record, final FlowNodeInstanceEntity entity) {
     final IncidentRecordValue recordValue = record.getValue();
-    entity
-        .setId(String.valueOf(recordValue.getElementInstanceKey()))
-        .setKey(recordValue.getElementInstanceKey())
-        .setPartitionId(record.getPartitionId())
-        .setFlowNodeId(recordValue.getElementId())
-        .setProcessInstanceKey(recordValue.getProcessInstanceKey())
-        .setProcessDefinitionKey(recordValue.getProcessDefinitionKey())
-        .setBpmnProcessId(recordValue.getBpmnProcessId())
-        .setTenantId(tenantOrDefault(recordValue.getTenantId()));
-
-    final String intentStr = record.getIntent().name();
-    if (intentStr.equals(IncidentIntent.CREATED.name())
-        || intentStr.equals(IncidentIntent.MIGRATED.name())) {
-      entity.setIncidentKey(record.getKey());
-    } else if (intentStr.equals(IncidentIntent.RESOLVED.name())) {
-      entity.setIncidentKey(null);
+    // build the treePath in the format
+    // <processInstanceKey>/<flowNodeInstanceKey>/.../<flowNodeInstanceKey>
+    // where upper level flowNodeInstanceKeys are normally subprocess(es) or multi-instance body
+    if (recordValue.getElementInstancePath() != null
+        && !recordValue.getElementInstancePath().isEmpty()) {
+      final Long currentProcessInstanceKey = Long.valueOf(entity.getId());
```

Comment:
❓ Specific reason why we not use the PI key from the record?

Reply:
Because its not the process instance key, but flow node instance key 🙈  I will rename the variable.